Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,391 @@
+   }
+   MainMove(); 
+}
+
+bool Projectile::CheckPassable()
+{
+    id_ptr_on<Projectile> p = this->GetId();
+    PassableLevel loc = GetPassable(GetDir());
+    if (loc != Passable::FULL)
+    {
+        SetPassable(GetDir(), Passable::FULL);
+    }
+    if (!CanPass(owner->GetPassable(GetDir()), passable_level))
+    {
+        owner->Bump(p);
+        if (loc != Passable::FULL)
+        {
+            SetPassable(GetDir(), loc);
+        }
+        return false;
+    }
+    if (loc != Passable::FULL)
+    {
+        SetPassable(GetDir(), loc);
+    }
+
+    auto neighbour = owner->GetNeighbour(GetDir());
+    if (   !CanPass(neighbour->GetPassable(D_ALL), passable_level)
+        || !CanPass(neighbour->GetPassable(helpers::revert_dir(GetDir())), passable_level))
+    {
+        neighbour->Bump(p);
+        return false;
+    }
+    
+    return true;
+}
+void Projectile::Bump(id_ptr_on<IMovable> item)
+{
+    id_ptr_on<Projectile> p = this->GetId();
+    item->Bump(p);
+    this->Delete();
+   
+//TODO maybe it can apply froce depending on bullet speed but that should be considered        ApplyForce(DirToVDir[m->dMove]);
+}
+void Projectile::MakeMovementPattern(VDir target,id_ptr_on<Human> someone)
+{
+    set_target_=1;
+    target_ = target;
+    int x = target.x;
+    int y = target.y;
+    int abs_x = std::abs(x);
+    int abs_y = std::abs(y);
+    shooter_ = someone;
+    Dir facing = someone->GetDir();
+    if(abs_x == abs_y)
+    {
+        if(y<0)
+        {
+            if(facing == 0|| facing == 1)
+            {
+                if(x>0)
+                {
+                    direction = D_RIGHT;
+                    cases_ = 3;
+                }
+                else
+                {
+                    direction = D_LEFT;
+                    cases_ = 6;
+                }
+            }
+            else
+            {
+                if(x>0)
+                {
+                    direction = D_UP;
+                    cases_ = 4;
+                }
+                else
+                {
+                    direction = D_UP;
+                    cases_ = 5;
+                }
+            }
+        }
+        else
+        {
+        if(facing == 0|| facing == 1)
+            {
+                if(x>0)
+                {
+                    direction = D_RIGHT;
+                    cases_ = 2;
+                }
+                else
+                {
+                    direction = D_LEFT;
+                    cases_ = 7;
+                }
+            }
+            else
+            {
+                if(x>0)
+                {
+                    direction = D_DOWN;
+                    cases_ = 1;
+                }
+                else
+                {
+                    direction = D_DOWN;
+                    cases_ = 8;
+                }
+            }
+        }
+    }
+    else if(x >= 0 && y >= 0)
+    {
+        if(x != 0)
+        {
+            if((float)y / (float)x > 1 )
+            {
```

Comment:
It is forbidden to use `float` or `double` in the core code (game logic). Moreover, you do not need it here at all.
`if ((float)y / (float)x > 1)` is equal to `if (y > x)` (well,  some exception for negarive numbers, but it should be handled separately).